tremor-pipeline/src/lib.rs

```diff
@@ -73,6 +77,51 @@ pub use op::{ConfigImpl, InitializableOperator, Operator};
 pub use tremor_script::prelude::EventOriginUri;
 pub(crate) type ExecPortIndexMap =
     HashMap<(usize, Cow<'static, str>), Vec<(usize, Cow<'static, str>)>>;
+#[derive(Debug, Clone)]
+///Pluggable-logging Appender
+pub struct PluggableLoggingAppender {
+    /// async Sender
+    pub tx: Sender<LoggingMsg>,
+}
+
+impl Append for PluggableLoggingAppender {
+    fn append(&self, record: &log::Record) -> anyhow::Result<()> {
+        let path: String;
```

Comment:
perhaps path could be Option<String> that way it's easily recognizable in the code as `null` or `string` to see if it was set :)

Reply:
nicely done 👍 code looks much simpler now too :D